packages/reactive-dict/reactive-dict.js

```diff
@@ -164,6 +164,23 @@ _.extend(ReactiveDict.prototype, {
     
   },
 
+  delete: function(key) {
+    var self      = this;
+    var oldKeys   = self.keys;
+    var didRemove = false;
+
+    if (_.has(oldKeys, key)) {
+      changed(self.keyDeps[key]);
+      changed(self.keyValueDeps[key][oldKeys[key]]);
+      changed(self.keyValueDeps[key]['undefined']);
+      delete self.keys[key]; // clean up
```

Comment:
right, I hadn't thought of that, we should probably clean up as much as possible.

`clear` should clean up as well, `clear`'s clean up should be a different gh issue, no?